storage/cleanup.go

```diff
@@ -132,38 +131,47 @@ func (m *cleanupManager) cleanupFilesetFiles(earliestToRetain time.Time) error {
 // writes and future writes, we need to shift flush time range by block size as the
 // time range for commit log files we need to check.
 func (m *cleanupManager) commitLogTimeRange(t time.Time) (time.Time, time.Time) {
-	flushStart, flushEnd := m.fm.FlushTimeStart(t), m.fm.FlushTimeEnd(t)
-	return flushStart.Add(-m.blockSize), flushEnd.Add(-m.blockSize)
+	var (
+		ropts      = m.opts.CommitLogOptions().RetentionOptions()
+		blockSize  = ropts.BlockSize()
+		flushStart = retention.FlushTimeStart(ropts, t)
+		flushEnd   = retention.FlushTimeEnd(ropts, t)
+	)
+	return flushStart.Add(-blockSize), flushEnd.Add(-blockSize)
```

Comment:
@robskillington weirdly, I see your comments for this line via email, but my GH webpage is missing them. Inlining for posterity:

> Hm, shouldn't flushEnd be flushEnd.Add(blockSize) not flushEnd.Add(-blockSize)?

Either way works, this way (same as original), is a little more paranoid so I left it as is. Will sync with you about this in person.